routes/drones.js

```diff
@@ -1,37 +1,83 @@
 const express = require('express');
 
-// require the Drone model here
+const Drone = require('../models/Drone.model')
 
 const router = express.Router();
 
 router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
+  Drone.find({})
+    .then((allDrones) =>
+      res.render('drones/list', {
+        allDrones
+      })
+    )
+    .catch((error) => `An error occurred when loading ${error}`)
 });
 
-router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
-});
+router.get("/drones/create", (req, res) => res.render("drones/create-form"));
 
 router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  const {
+    name,
+    propellors,
+    maxSpeed
+  } = req.body;
+
+  Drone.create({
+      name,
+      propellors,
+      maxSpeed
+    }).then(() => res.redirect("/drones"))
+    .catch((error => {
+      `An error occurred while creating ${error}`,
+      res.render('drones/create-form')
```

Comment:
You have a `error.hbs` file in your views folder, it would be more appropriate to render that page instead of `create-form.hbs`.

You could even pass the error to the page `error.hbs` so the user can be informed as to what the problem is
```
res.render('error', {error});

Reply:
Ok, I could change that but just so you know, the lab tells you to re-render the create view again, same in case of the update

Reply:
I know, you don't have to change it. 

I was just making you aware that this is a possible way of using your a view file. Most Node applications have some kind of middleware in place for error handling so that when there's a problem it can be displayed to the user/developer(there's an [extensive list of ways](https://expressjs.com/en/guide/error-handling.html) to do that).

It's beyond the scope of this assignment to be honest, I just felt like mentioning it for the sake of sharing knowledge.

Sorry for the confusion.

Reply:
Ahh, like that, got it! Thanks for the tip Lloyd!